File: mixed_pub_sub/mixed_pub_sub/twist_publisher.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from geometry_msgs.msg import TwistStamped


class TwistPublisher(Node):
    def __init__(self):
        super().__init__('twist_publisher')
        self.publisher_ = self.create_publisher(TwistStamped, 'cmd_vel', 10)
        self.timer = self.create_timer(0.1, self.publish_twist)

    def publish_twist(self):
        msg = TwistStamped()
        msg.header.stamp = self.get_clock().now().to_msg()
        msg.twist.linear.x = 1.0
        msg.twist.angular.z = 0.5
        self.publisher_.publish(msg)
        self.get_logger().info('Publishing TwistStamped message')


def main(args=None):
    rclpy.init(args=args)
    node = TwistPublisher()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: mixed_cpp_py_pkg/src/twist_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <geometry_msgs/msg/twist_stamped.hpp>

class TwistPublisher : public rclcpp::Node
{
public:
    TwistPublisher() : Node("twist_publisher")
    {
        publisher_ = this->create_publisher<geometry_msgs::msg::TwistStamped>("cmd_vel", 10);
        timer_ = this->create_wall_timer(
            std::chrono::milliseconds(100),
            std::bind(&TwistPublisher::publish_twist, this));
    }

private:
    void publish_twist()
    {
        auto msg = geometry_msgs::msg::TwistStamped();
        msg.header.stamp = this->get_clock()->now();
        msg.twist.linear.x = 1.0;
        msg.twist.angular.z = 0.5;
        publisher_->publish(msg);
        RCLCPP_INFO(this->get_logger(), "Publishing TwistStamped message");
    }
    rclcpp::Publisher<geometry_msgs::msg::TwistStamped>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<TwistPublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}